### PR TITLE
Updated analysis: medulloblastoma consensus subtypes

### PR DESCRIPTION
**Pull request by komalsrathi**
#### What scientific question is your analysis addressing?

Classifier accuracy was re-evaluated with an updated pathology file. To assign molecular subtypes, a consensus was taken from the predictions of the two classifiers: medulloPackage and MM2S.

For the consensus output using batch-corrected input data, 29/43 (67.44%) and for the consensus output using uncorrected data, 30/43 (69.76%) match with the reported subtype. The difference between the two was that `BS_V96WVE3Z` is correctly predicted as SHH by consensus uncorrected output but not by batch-corrected output. Further digging shows that medulloPackage correctly predicts BS_V96WVE3Z as SHH using both batch-corrected and uncorrected input data but MM2S incorrectly predicts this sample as Group3 using batch-corrected input and hence no consensus was obtained for this sample using batch-corrected data.  

The following table shows the difference between the two outputs:

| Consensus Output | Kids_First_Biospecimen_ID | pathology_subtype | MM2S_best_fit | medulloPackage_best_fit | molecular_subtype | match |
|------------------|---------------------------|-------------------|---------------|-------------------------|-------------------|-------|
| Uncorrected      | BS_V96WVE3Z               | SHH               | SHH           | SHH                     | SHH               | TRUE  |
| Corrected        | BS_V96WVE3Z               | SHH               | Group3        | SHH                     | SHH                | NA    |

#### What was your approach?

1. Updated v17 histologies to v18
2. Used https://github.com/AlexsLemonade/OpenPBTA-analysis/files/6338682/openPBTA-mb-pathology-subtypes.csv provided by @jharenza -> converted to rds and used that instead of `input/expected_class_v17.rds`

#### What GitHub issue does your pull request address?

https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/747

#### Which areas should receive a particularly close look?

N/A

#### Is there anything that you want to discuss further?

N/A

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

Table

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Review by jharenza — CHANGES_REQUESTED** [review]
Hi @komalsrathi - thanks for this update.

I noticed a few things that I think we should update. 

1. For `pathology_subtype == non-WNT`, if there is a subgroup predicted by a classifier as non-WNT (SHH, Group3, or Group4), you are saying this prediction is `false`, but I think we need to update those to `true`. Will you update this and update the accuracies? I am interested to see whether there is a strong difference in classifiers then or if they still remain on par with each other.
2. For cases in which the consensus does not match, but we have a `pathology_subtype`, I think we should add the `pathology_subtype` as the molecular_subtype. 

Just a general comment that I also found the accuracy dropped when using batch corrected consensus, which helps justify the use of uncorrected data here.

**Comment by komalsrathi** [comment]
@jharenza updated both, thanks!

**Comment by komalsrathi** [comment]
@jharenza updated both, thanks!

File: analyses/molecular-subtyping-MB/README.md [hunk 1]
```diff
@@ -143,14 +143,7 @@ For each sample_id with multiple RNA samples, the following logic is implemented
...
 
-For both the consensus corrected output and consensus uncorrected output, 25/32 match with the reported subtype so the % accuracy is the same i.e. 78.125%. Between the two consensus outputs, there is an overlap of 24/25 matched subtypes. The following samples are mismatched between the two consensus outputs:
-
-BS_HB03GSHF is correctly predicted by consensus batch-corrected output but not by uncorrected output:
-
-| Consensus Output | Kids_First_Biospecimen_ID | pathology_subtype | MM2S_best_fit | medulloPackage_best_fit | molecular_subtype | match |
-|------------------|---------------------------|-------------------|---------------|-------------------------|-------------------|-------|
-| Corrected        | BS_HB03GSHF               | WNT               | WNT           | WNT                     | WNT               | TRUE  |
-| Uncorrected      | BS_HB03GSHF               | WNT               | SHH           | WNT                     | NA                | NA    |
+For the consensus corrected output, 29/43 (67.44%) and for the consensus uncorrected output, 30/43 (69.76%) match with the reported subtype. The following table shows the difference between the two outputs:
```
Comment (jharenza) [inline analyses/molecular-subtyping-MB/README.md:146]: @komalsrathi is this still correct? It seems to be that if each individual classifier has 90 or 95% accuracy alone, that we should not see such a drastic drop when combining them.

**Review by jharenza — CHANGES_REQUESTED** [review]
Hi @komalsrathi thanks for the changes and the classifiers now look to be on par with their respective publications, which is great!

When I mentioned 
>For cases in which the consensus does not match, but we have a pathology_subtype, I think we should add the pathology_subtype as the molecular_subtype.

I neglected to think about the behavior when the pathology reports are not exact and I think that in these cases, we should only keep the pathology subtype when it is confident in one single subtype. Sorry about that - will you make that change?

Also made a comment about combined accuracy - think that may need an update as well?

File: analyses/molecular-subtyping-MB/README.md (hunk 1 above)
Comment (komalsrathi) [reply analyses/molecular-subtyping-MB/README.md]: @jharenza thanks for catching that - I forgot to update the README.

**Review by jharenza — APPROVED** [review]
This looks great now, thank you!

**Comment by komalsrathi** [comment]
@jharenza can you please check if this can be merged? Thanks!

**Comment by jharenza** [comment]
> @jharenza can you please check if this can be merged? Thanks! 

I realized that with v19, we have added the mb pathology subtype file. Will you point to that now instead of the input RDS? Then we can merge. Thanks!

**Comment by jharenza** [comment]
> @jharenza can you point me to that file - sorry looked under data/release-v19-20210423 but maybe missing something (probably need more coffee lol)

Ahh, it looks like we missed it in the md5sum file, so it is not being downloaded in this release. I will merge this once checks pass and we will add to v20.

**Pull request merged**